lldb/source/Commands/CommandObjectDWIMPrint.cpp

```diff
@@ -61,6 +63,36 @@ void CommandObjectDWIMPrint::HandleArgumentCompletion(
 
 bool CommandObjectDWIMPrint::DoExecute(StringRef command,
                                        CommandReturnObject &result) {
+  // BEGIN SWIFT
+  // Dump the contents of the valobj to result, possibly prepending it with
+  // a note telling the user that p is potentially better.
+  auto write_valobj_to_result =
+      [&](ValueObject &valobj, const DumpValueObjectOptions &dump_options) {
+        // Identify the default output of po "<Name: 0x...>.
+        // The regex is:
+        // - Start with "<".
+        // - Followed by 1 or more non-whitespace characters.
+        // - Followed by ": 0x".
+        // - Followed by 1 or more characters in the 0-9 or a-f range.
+        // - Followed by ">".
+        static const std::regex swift_class_regex("^<[^\\s]+: 0x[0-9a-f]+>");
+        StreamString temp_result_stream;
+        valobj.Dump(temp_result_stream, dump_options);
+
+        bool is_po = m_varobj_options.use_objc;
+        if (is_po && std::regex_search(temp_result_stream.GetData(),
+                                       swift_class_regex)) {
+          result.AppendMessage(
+              "note: object description requested, but type doesn't implement "
+              "a custom object description. "
+              "Consider using the default formatter (typically aliased to "
+              "\"p\") "
```

Comment:
I think we can assume the default `p`. If they've re-aliased `p`, they can probably figure out the right meaning.

Reply:
also the formatting of this string is strange (some short lines some long lines)

Reply:
What about `Consider using "p" instead of "po"`?

Reply:
Both users that use `expr -O` probably can figure it out ;-)